docs/sources/reference/components/prometheus/prometheus.remote.queue.md

```diff
@@ -0,0 +1,296 @@
+[Prometheus Remote Write protocol][remote_write-spec].
+
+You can specify multiple `prometheus.remote.queue` components by giving them different labels.
+
+You should consider everything here extremely experimental and highly subject to change.
+[emote_write-spec]: https://docs.google.com/document/d/1LPhVRSFkGNSuU1fBd81ulhsCPR4hkSZyyBj1SZ8fWOM/edit
+
+  
+
+## Usage
+
+```alloy
+prometheus.remote.queue "LABEL" {
+  endpoint "default "{
+    url = REMOTE_WRITE_URL
+
+    ...
+  }
+
+  ...
+}
+```
+
+## Arguments
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`ttl` | `time` | `duration` | How long the timestamp of a signal is valid before the signal is discarded. | `2h` | no
+
+## Blocks
+
+The following blocks are supported inside the definition of
+`prometheus.remote.queue`:
+
+Hierarchy | Block | Description | Required
+--------- | ----- | ----------- | --------
+serialization | [serialization][] | Configuration for serializing and writing to disk | no
+endpoint | [endpoint][] | Location to send metrics to. | no
+endpoint > basic_auth | [basic_auth][] | Configure basic_auth for authenticating to the endpoint. | no
+
+The `>` symbol indicates deeper levels of nesting. For example, `endpoint >
+basic_auth` refers to a `basic_auth` block defined inside an
+`endpoint` block.
+
+[endpoint]: #endpoint-block
+[basic_auth]: #basic_auth-block
+[serialization]: #serialization-block
+
+### serialization block
+
+The `serialization` block describes how often and at what limits to write to disk. Serialization settings
+are shared for each `endpoint.`
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`max_signals_to_batch` | `uint` | The maximum number of signals before they are batched to disk. | `10,000` | no
+`batch_frequency` | `duration` | How often to batch signals to disk if `max_signals_to_batch` is not reached. | no
+
+
+### endpoint block
+
+The `endpoint` block describes a single location to send metrics to. Multiple
+`endpoint` blocks can be provided to send metrics to multiple locations. Each
+`endpoint` will have its own WAL folder
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`url` | `string` | Full URL to send metrics to. | | yes
+`name` | `string` | Optional name to identify the endpoint in metrics. | | no
+`write_timeout` | `duration` | Timeout for requests made to the URL. | `"30s"` | no
+`retry_backoff` | `duration` | How often to wait between retries. | `1s` | no
+`max_retry_backoff_attempts` | Maximum number of retries before dropping the batch. | `1s` | no
+`batch_count` | `uint` | How many series to queue in each queue. | `1,000` | no
+`flush_frequency` | `duration` | How often to wait until sending if `batch_count` is not trigger. | `1s` | no
```

Comment:
```suggestion
`flush_interval` | `duration` | How often to wait until sending if `batch_count` is not trigger. | `1s` | no
```

frequency is `1/interval` and would be in Hz

Reply:
We use both interval and frequency all over the configuration. The original remote_write uses both, but both use the verbiage frequency/frequent to describe the value so it felt better. 

remote write below
`truncate_frequency` | `duration` | How frequently to clean up the WAL. | `"2h"` | no
`send_interval` | `duration` | How frequently metric metadata is sent to the endpoint. | `"1m"` | no

Reply:
haha, I see... well, we can improve on that, but I guess it's a nit

Reply:
Either way works if it's consistent with upstream and within Alloy.  But... we can look at the top level definitions...

* frequency... is the rate something happens or the number of times something happens within a given time period
* interval... is a measurable time between things or events

So what are we providing? A fixed time.. for example, 30 seconds? Then it's an interval. if it's something like like 150/hour then it's a frequency.  

In this case... I'd probably opt to go with 

```suggestion
`flush_interval` | `duration` | How long to wait until sending if `batch_count` is not triggered. | `1s` | no
```